### PR TITLE
https://blueprints.launchpad.net/or/+spec/wheel-bearing-impacts - mov…

### PR DESCRIPTION
**Pull request by peternewell**
…e wheel bearing temperature functionality into MS

### CODE REVIEW

File: Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs [hunk 1]
```diff
@@ -45,6 +45,7 @@
 using ORTS.Scripting.Api;
 using ORTS.Settings;
 using System;
+using Orts.Parsers.Msts;
```
Comment (Csantucci) [inline Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs]: Please add this line in alphabetic order.

File: Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs [hunk 2]
```diff
@@ -176,6 +180,79 @@ public abstract class TrainCar
...
+        public float ActivityElapsedDurationS;
+        public float HotBoxStartTimeS;
+
+        // Setup for ambient temperature dependency
+        Interpolator OutsideWinterTempbyLatitudeC;
+        Interpolator OutsideAutumnTempbyLatitudeC;
+        Interpolator OutsideSpringTempbyLatitudeC;
+        Interpolator OutsideSummerTempbyLatitudeC;
+        public bool AmbientTemperatureInitialised;
+
```
Comment (Csantucci) [inline Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs]: Please describe the above variables adding a comment to each of them.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs [hunk 3]
```diff
@@ -1660,105 +1665,185 @@ private void UpdateTrainBaseResistance()
...
-                    {
-                        IsSnowing = true;  // Weather snowing - freezing conditions
-                    }
 
                     // Dtermine the starting friction factor based upon the type of bearing
 
-                    float StartFrictionLow = 0.0f;
-                    float StartFrictionHigh = 0.0f;
+                    float StartFrictionLowLoad = 0.0f;
+                    float StartFrictionHighLoad = 0.0f;
```
Comment (Csantucci) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs]: Please describe the two above variables.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs [hunk 4]
```diff
@@ -1660,105 +1665,185 @@ private void UpdateTrainBaseResistance()
...
                         else if (Kg.ToTUS(MassKG) > 100.0)
                         {
-                            StaticFrictionFactorLb = StartFrictionHigh;  // Starting friction for a > 100 ton(US) car with standard roller bearings
+                            StaticFrictionFactorLb = StartFrictionHighLoad;  // Starting friction for a > 100 ton(US) car with standard roller bearings
                         }
                         else
                         {
-                            StaticFrictionFactorLb = (((Kg.ToTUS(MassKG) - 10.0f) / 90.0f) * (StartFrictionHigh - StartFrictionLow)) + StartFrictionLow;
+                            StaticFrictionFactorLb = (((Kg.ToTUS(MassKG) - 10.0f) / 90.0f) * (StartFrictionHighLoad - StartFrictionLowLoad)) + StartFrictionLowLoad;
                         }
```
Comment (Csantucci) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs]: This should be brought to metric units here, which is default for OR, if there aren't strong reasons against.

File: Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs (hunk 1 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs]: Ok, done.

File: Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs (hunk 2 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs]: Ok, done.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs (hunk 3 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs]: Ok, done.

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs (hunk 4 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs]: Ok, converted.

**Pull request merged**